sky/exceptions.py

```diff
@@ -10,9 +11,18 @@
 class ResourcesUnavailableError(Exception):
     """Raised when resources are unavailable."""
 
-    def __init__(self, *args: object, no_failover: bool = False) -> None:
+    def __init__(
+            self,
+            *args: object,
+            no_failover: bool = False,
+            failover_reasons: Optional[Dict[Type[Exception],
+                                            str]] = None) -> None:
         super().__init__(*args)
         self.no_failover = no_failover
+        # Mapping from exception type to reason for failover.
+        if failover_reasons is None:
+            failover_reasons = dict()
+        self.failover_reasons: Dict[Type[Exception], str] = failover_reasons
```

Comment:
Document when this will be = 0, = 1, >1.

Reply:
Refactored this to be only checking whether there is `ResourcesUnavailableError` in the `failover_reasons`. PTAL. : )